core/types.go

```diff
@@ -260,14 +260,38 @@ type RenewalInfo struct {
 // using a very simple renewal calculation: calculate a point 2/3rds of the way
 // through the validity period, then give a 2-day window around that. Both the
 // `issued` and `expires` timestamps are expected to be UTC.
-func RenewalInfoSimple(issued time.Time, expires time.Time) *RenewalInfo {
+func RenewalInfoSimple(issued time.Time, expires time.Time, now time.Time) *RenewalInfo {
 	validity := expires.Add(time.Second).Sub(issued)
 	renewalOffset := validity / time.Duration(3)
 	idealRenewal := expires.Add(-renewalOffset)
+	windowStart := idealRenewal.Add(-24 * time.Hour)
+	windowEnd := idealRenewal.Add(24 * time.Hour)
+
+	if expires.Before(now) {
+		// If the cert has expired, return `RenewalInfoImmediate` instead.
+		return RenewalInfoImmediate(now)
+	}
+	
+	// Ensure RenewalWindow is not after the expiry
+	if windowEnd.After(expires){
```

Comment:
You still have formatting issues here, and below. You should be running `gofmt` automatically.
```suggestion
	if windowEnd.After(expires) {
```

Reply:
Ah! It's not performing on directories, just specific files.  Fixed.